### ISSUE TITLE
Release version `2023.1.0`

### ISSUE THREAD

**Issue by namurphy** [PlasmaPy/PlasmaPy#1878]
We're planning to do our first date-based release of PlasmaPy, likely in the next week or so.

I'll update this more later...

**Comment by StanczakDominik** [comment]
And that's done!

Three ideas for follow up:
- [ ] have an issue template for the release checklists?
- [ ] drop the pre-commit update step, it's slow and irrelevant to the release process anyway
- [ ] fix the script that bumps the changelog entry in `docs/whatsnew/index.rst` to skip doing that on rc versions, or possibly find and remove them once we push the full release
- [ ] check again whether #1340 would help

**Comment by namurphy** [comment]
> drop the pre-commit update step, it's slow and irrelevant to the release process anyway

Agreed!  My original motivations in adding this to the release process were:
 - Making sure we update pre-commit hooks at all
 - Applying major formatting changes so that it's easier to do a backport to the previous release, if needed
The first one no longer matters because we now have pre-commit autoupdates like in #1863, and the second one is less relevant because black seems to have stabilized. So, I'm happy to drop this step.

**Comment by StanczakDominik** [comment]
Also remaining to do:

- [ ] announcements
    - [ ] newsletter
    - [ ] fb, linkedin, twitter, mastodon?
    - [ ] APS DPP Engage
- [ ] ensure conda-forge (should happen automatically by tomorrow)
- [ ] cleanup release guide accordingly
